part-1.py

```diff
@@ -4,17 +4,48 @@
 # the appropriate comment.
 
 # factorial
-
+def factorial(n):
+        if n < 0:
+            raise ValueError("ValueError exception thrown")
+        if n == 0 :
+            return 1
+        else:
+            return n * factorial(n -1 )
+    
+    
 
 
 # reverse
-
+def reverse(text):
+    if text:
+        text = text[::-1]
+        return text
+    if text == "":
+        return ""
```

Comment:
So you have a base case and an incrementer, but you aren't recursively calling `reverse()`, which is necessary for recursion